fs.c

```diff
@@ -1903,6 +1903,18 @@ static int FS_ChooseUserDir(userdirmode_t userdirmode, char *userdir, size_t use
 		if(homedir)
 		{
 			dpsnprintf(userdir, userdirsize, "%s/.%s/", homedir, gameuserdirname);
+#if defined(__unix__) || defined(__unix)
+			// Check for darkplaces config paths in this order: ~/.darkplaces, XDG_CONFIG_HOME/darkplaces, ~/.config/darkplaces.
```

Comment:
This encodes a preference order. It prefers ~/.darkplaces if it exists, but creates ~/.config/darkplaces if nothing exists. Please explain this in the comment.